prusti-server/src/process_verification.rs

```diff
@@ -10,67 +10,70 @@ use prusti_common::{config, report::log::report, vir::ToViper, Stopwatch};
 use std::{fs::create_dir_all, path::PathBuf};
 use viper::{Cache, VerificationBackend, VerificationContext};
 
-pub fn process_verification_request_cache<'v, 't: 'v>(
+pub fn process_verification_request<'v, 't: 'v>(
     verification_context: &'v VerificationContext<'t>,
     request: VerificationRequest,
     cache: impl Cache,
 ) -> viper::VerificationResult {
-    if !config::enable_cache() || config::print_hash() {
-        process_verification_request(verification_context, request)
-    } else {
+    let ast_utils = verification_context.new_ast_utils();
+    ast_utils.with_local_frame(16, || {
```

Comment:
I'm not sure how expensive `with_local_frame` is? If it's cheap then this is fine, but if not then it's problematic for the common case where `!config::dump_viper_program()` but we get a cache hit

Reply:
I've not benchmarked it, but it should be cheaper than doing a method call in the JVM.

Reply:
(avoided)